appcues/src/main/java/com/appcues/debugger/AppcuesDebuggerManager.kt

```diff
@@ -90,88 +85,84 @@ internal class AppcuesDebuggerManager(
         debuggerViewModel?.reset()
     }
 
+    override fun onActivityCreated(activity: Activity, savedInstanceState: Bundle?) = Unit
+    override fun onActivityStarted(activity: Activity) = Unit
     override fun onActivityResumed(activity: Activity) {
-        this.currentActivity = activity
-        debuggerViewModel?.let { addDebuggerView(activity, it) }
-    }
-
-    override fun onActivityPostResumed(activity: Activity) {
-        // not sure if this is necessary but I wanted to make sure we register our back press after everyone else
-        // in case customer is using fragments with the same approach then our callback must be the last one so we can better
-        // control native back press by enabling and disabling on our side
-        setDebuggerBackPressCallback(activity)
-        // also to make sure that if they change fragment, we will register again after the new fragment is attached
-        (activity as? FragmentActivity)?.supportFragmentManager?.addFragmentOnAttachListener { _, _ ->
-            setDebuggerBackPressCallback(activity)
+        currentActivity = activity
+        // setting this on postDelayed ensures we wait for
+        // any transition from one activity to the another before we actually try to add
+        // debugger view
+        debuggerViewModel?.let {
+            Handler(Looper.getMainLooper()).postDelayed({ addDebuggerView(it) }, 0)
         }
     }
 
     override fun onActivityPaused(activity: Activity) = Unit
-    override fun onActivityStarted(activity: Activity) = Unit
     override fun onActivityStopped(activity: Activity) = Unit
-    override fun onActivityCreated(activity: Activity, bundle: Bundle?) = Unit
-    override fun onActivitySaveInstanceState(activity: Activity, bundle: Bundle) = Unit
+    override fun onActivitySaveInstanceState(activity: Activity, outState: Bundle) = Unit
     override fun onActivityDestroyed(activity: Activity) = Unit
 
-    private fun addDebuggerView(activity: Activity, debuggerViewModel: DebuggerViewModel) {
-        getParentView(activity).also {
-            // if view is not there
-            if (it.findViewById<ComposeView>(R.id.appcues_debugger_view) == null) {
-                // then we add
-                it.addView(
-                    ComposeView(activity).apply {
+    private fun addDebuggerView(debuggerViewModel: DebuggerViewModel) {
```

Comment:
changes to this method to make it more readable and in sync with removeDebuggerView.  both dependent on "currentActivity"